src/components/hoc/RestrictedRoute.js

```diff
@@ -0,0 +1,27 @@
+import React from 'react';
+import { Route, Redirect } from 'react-router';
+
+export default function RestrictedRoute({
+  isAllowed = false,
+  component: Component,
+  render: RenderedComponent,
+  redirectTo,
+  ...routeProps
+}) {
+  return (
+    <Route
+      {...routeProps}
+      render={(props) => {
+        if (isAllowed) {
+          // Compatibility with both Route render or Route component
+          if (!!RenderedComponent) {
```

Comment:
why the double negation?

Reply:
To make sure its result is a `boolean`. Not necessary. Will change. 👍